test/integration/tests/pcrallocate.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+#;**********************************************************************;
+#
+# Copyright (c) 2016-2018, Intel Corporation
+# Copyright (c) 2019, Fraunhofer SIT
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# 3. Neither the name of Intel Corporation nor the names of its contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+# THE POSSIBILITY OF SUCH DAMAGE.
+#;**********************************************************************;
+
+source helpers.sh
+
+cleanup() {
+    rm -f pcrs.out
+
+    if [ "$1" != "no-shut-down" ]; then
+          shut_down
+    fi
+}
+trap cleanup EXIT
+
+start_up
+
+cleanup "no-shut-down"
+
+tpm2_pcrallocate -P "" sha1:7,8,9,10,16,17,18,19+sha256:0,1,2,3,4,5,6
+
+tpm2_pcrallocate
+
+tpm2_pcrallocate sha1:all+sha256:all+sha384:all+sha512:all
```

Comment:
Ok, I'm adding the yaml helpers. The current output is yaml compliant though:
```
>>> yaml.load("""SelectedPCRs:
...   - bank: sha1
...     pcrs: [ 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]
...   - bank: sha256
...     pcrs: [ 0, 1, 2, 3, 4, 5, 6, 7 ]
... """)
{'SelectedPCRs': [{'bank': 'sha1', 'pcrs': [8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23]}, {'bank': 'sha256', 'pcrs': [0, 1, 2, 3, 4, 5, 6, 7]}]}
```
But I take it, you'd like a different format ? Which one ? Like this ?
```
SelectedPCRs:
  - sha1: [ 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]
  - sha256: [ 0, 1, 2, 3, 4, 5, 6, 7 ]
```
Or rather without the `selectedPCRs` thing ?

Reply:
I like this one with a minor tweak:
```YAML
selected-pcrs:
  - sha1: [ 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23 ]
  - sha256: [ 0, 1, 2, 3, 4, 5, 6, 7 ]
```
Theirs an existing tool called tpm2_listpcrs. It should probably go away as tpm2_getcap supports listing pcrs, or we need to understand what the feature delta is and close it. At a minimum, that tpm2_pcrlist tool should be using the same YAML output routine that you introduced as to not duplicate it.

Reply:
tpm2_pcrlist should probably be called tpm2_pcrread since it just reads all pcrs.
That's the feature-delta... ;-)

I'll update the print output accordingly.